src/main/java/io/nats/client/impl/NatsConnectionWriter.java

```diff
@@ -105,10 +106,7 @@ Future<Boolean> stop() {
             try {
                 this.outgoing.pause();
                 this.reconnectOutgoing.pause();
-                // Clear old ping/pong requests
-                this.outgoing.filter((msg) ->
-                    msg.isProtocol() &&
-                        (msg.getProtocolBab().equals(OP_PING_BYTES) || msg.getProtocolBab().equals(OP_PONG_BYTES)));
+                this.outgoing.filter(NatsMessage::isProtocolFilterOnStop);

```

Comment:
Leverage the new flag for faster filtering instead of doing byte compares on proto bytes